server/src/models/timestamps.rs

```diff
@@ -0,0 +1,352 @@
+            }
+        }
+        last_delta = delta;
+        last_timestamp = *timestamp;
+    }
+
+    // All remaining bits in the byte the BitVecBuilder is currently packing
+    // bits into is set to one to indicate that all timestamps are decompressed.
+    compressed_timestamps.finish_with_one_bits()
+}
+
+/// Decompress all of a segment's timestamps which are compressed as
+/// `start_time` for segments of length one, `start_time` and `end_time` for
+/// segments of length two, the segment's length for regular time series, or
+/// using Gorilla's compression method for timestamps for irregular time series.
+pub fn decompress_all_timestamps(
+    start_time: Timestamp,
+    end_time: Timestamp,
+    residual_timestamps: &[u8],
+) -> TimestampArray {
+    if residual_timestamps.is_empty() && start_time == end_time {
+        // Timestamps are assumed to be unique so the segment has one timestamp.
+        let mut timestamp_builder = TimestampBuilder::new(1);
+        timestamp_builder.append_value(start_time);
+        timestamp_builder.finish()
+    } else if residual_timestamps.is_empty() {
+        // Timestamps are assumed to be unique so the segment has two timestamp.
+        let mut timestamp_builder = TimestampBuilder::new(2);
+        timestamp_builder.append_value(start_time);
+        timestamp_builder.append_value(end_time);
+        timestamp_builder.finish()
+    } else if residual_timestamps[0] & 128 == 0 {
+        // The flag bit is zero, so only the segment's length is stored as an
+        // integer with all the prefix zeros stripped from the integer.
+        decompress_all_regular_timestamps(start_time, end_time, residual_timestamps)
+    } else {
+        // The flag bit is one, so the timestamps are compressed as
+        // delta-of-deltas stored using a variable length binary encoding.
+        decompress_all_irregular_timestamps(start_time, end_time, residual_timestamps)
+    }
+}
+
+/// Decompress all of a segment's timestamps, which for this segment are sampled
+/// at a regular sampling interval, and thus compressed as the segment's length.
+fn decompress_all_regular_timestamps(
+    start_time: Timestamp,
+    end_time: Timestamp,
+    residual_timestamps: &[u8],
+) -> TimestampArray {
+    let mut bytes_to_decode = [0; 8];
+    bytes_to_decode[..residual_timestamps.len()].copy_from_slice(residual_timestamps);
+    let length = usize::from_le_bytes(bytes_to_decode);
+    let sampling_interval = (end_time - start_time) as usize / (length - 1);
+    let mut timestamp_builder = TimestampBuilder::new(length);
+    for timestamp in (start_time..=end_time).step_by(sampling_interval) {
+        timestamp_builder.append_value(timestamp);
+    }
+    timestamp_builder.finish()
+}
+
+/// Decompress all of a segment's timestamps, which for this segment are sampled
+/// at an irregular sampling interval, and thus compressed using Gorilla's
+/// compression method for timestamps for irregular time series.
```

Comment:
Maybe "for timestamps in irregular time series." instead.

Reply:
Fixed by removing "... for irregular time series".